### PR TITLE
ci: integrate code coverage reporting

### PR DESCRIPTION
**Pull request by lavarou**
Update `test-pull-request` workflow with steps to generate [`gcov`](https://gcc.gnu.org/onlinedocs/gcc/Gcov.html) code coverage reports for `agent` and `axiom` and upload them to [codecov.io](https://app.codecov.io/gh/newrelic/newrelic-php-agent). Since there are no differences between `gnu` and `musl` platforms as well as `amd64` and `arm64` architectures, reports are only generated for `gnu` on `amd64` for all supported PHP versions.

### CODE REVIEW

File: axiom/nr_daemon_spawn.c [hunk 1]
```diff
@@ -144,7 +144,8 @@ nr_argv_t* nr_daemon_args_to_argv(const char* name,
 }
 
 #ifdef DO_GCOV
-extern void __gcov_flush(void);
+extern void __gcov_reset(void);
```
Comment (zsistla) [inline axiom/nr_daemon_spawn.c:147]: I'm curious why this change?

Comment (lavarou) [reply axiom/nr_daemon_spawn.c:147]: See 92eadcf8519e8c9b20bdb3c6bd8ce8029a3fc074.

**Review by zsistla — APPROVED** [review]
Great work, excited for this to go in!

**Review by ZNeumann — APPROVED** [review]
